File: backend/modules/email_workflow.py
"""
Email Workflow Manager - Gerencia o fluxo de aprovação de emails
Controla as etapas: recebido → analisado → aprovado → pronto → publicado
"""

import logging
from typing import Dict, List, Optional, Any
from datetime import datetime, timedelta
from enum import Enum
import json

import sys
import os

# Adicionar diretório pai ao path
backend_dir = os.path.dirname(os.path.dirname(__file__))
if backend_dir not in sys.path:
    sys.path.insert(0, backend_dir)

try:
    # Tentar import relativo primeiro (produção)
    from ..database import db
    from .ai_processor import ai_processor
    from .wordpress_publisher import wp_publisher
except ImportError:
    # Fallback para import absoluto (desenvolvimento)
    from database import db
    from modules.ai_processor import ai_processor
    from modules.wordpress_publisher import wp_publisher

logger = logging.getLogger(__name__)

class WorkflowStage(Enum):
    RECEIVED = "received"
    ANALYZED = "analyzed"
    APPROVED_CONTENT = "approved_content"
    READY_PUBLISH = "ready_publish"
    PUBLISHED = "published"
    REJECTED = "rejected"
    ARCHIVED = "archived"

class Priority(Enum):
    HIGH = 1
    MEDIUM = 2
    LOW = 3

class EmailWorkflowManager:
    """Gerenciador do workflow de aprovação de emails"""
    
    def __init__(self):
        self.db = db
        logger.info("📋 Email Workflow Manager inicializado")
    
    # ==========================================
    # DASHBOARD E LISTAGEM
    # ==========================================
    
    def get_dashboard_stats(self, user_id: Optional[str] = None, days_back: int = 30) -> Dict:
        """Obter estatísticas para o dashboard"""
        try:
            result = self.db.client.rpc('get_email_workflow_stats', {
                'p_user_id': user_id,
                'p_days_back': days_back
            }).execute()
            
            if result.data:
                return result.data[0] if result.data else {}
            
            return {}
            
        except Exception as e:
            logger.error(f"Erro ao obter stats do dashboard: {e}")
            return {}
    
    def list_emails(self, 
                   stage: Optional[str] = None,
                   priority: Optional[int] = None,
                   assigned_to: Optional[str] = None,
                   is_auto_process: Optional[bool] = None,
                   limit: int = 50,
                   offset: int = 0) -> Dict:
        """Listar emails com filtros"""
        try:
            query = self.db.client.from_("email_dashboard").select("*")
            
            if stage:
                query = query.eq("workflow_stage", stage)
            if priority:
                query = query.eq("priority", priority)
            if assigned_to:
                query = query.eq("assigned_to", assigned_to)
            if is_auto_process is not None:
                query = query.eq("is_auto_process", is_auto_process)
            
            result = query.range(offset, offset + limit - 1).execute()
            
            return {
                "emails": result.data if result.data else [],
                "total": len(result.data) if result.data else 0
            }
            
        except Exception as e:
            logger.error(f"Erro ao listar emails: {e}")
            return {"emails": [], "total": 0}
    
    def get_email_details(self, email_id: str) -> Optional[Dict]:
        """Obter detalhes completos de um email"""
        try:
            # Buscar email
            email_result = self.db.client.from_("email_cache").select("*").eq("id", email_id).single().execute()
            
            if not email_result.data:
                return None
            
            email = email_result.data
            
            # Buscar histórico
            history_result = self.db.client.from_("email_workflow_history").select("*").eq("email_id", email_id).order("created_at", desc=True).execute()
            
            email["history"] = history_result.data if history_result.data else []
            
            return email
            
        except Exception as e:
            logger.error(f"Erro ao obter detalhes do email: {e}")
            return None
    
    # ==========================================
    # PROCESSAMENTO E ANÁLISE
    # ==========================================
    
    def analyze_email(self, email_id: str, user_id: str) -> Dict:
        """Analisar email com IA (primeira etapa do workflow)"""
        try:
            # Buscar email
            email = self.get_email_details(email_id)
            if not email:
                return {"success": False, "error": "Email não encontrado"}
            
            if email["workflow_stage"] != WorkflowStage.RECEIVED.value:
                return {"success": False, "error": "Email já foi analisado"}
            
            # Analisar com IA
            content = f"{email['subject']}\n\n{email['content_text']}"
            
            # Análise de categoria e relevância
            analysis = ai_processor.analyze_email_content(content)
            
            # Gerar conteúdo inicial se relevante
            generated_content = None
            if analysis.get("is_relevant", False):
                generated_content = ai_processor.generate_content_from_email(content)
            
            # Preparar dados da análise
            ai_analysis = {
                "category": analysis.get("category", "geral"),
                "confidence": analysis.get("confidence", 0.5),
                "is_relevant": analysis.get("is_relevant", False),
                "topics": analysis.get("topics", []),
                "generated_content": generated_content,
                "analysis_date": datetime.now().isoformat()
            }
            
            # Atualizar email
            update_result = self.db.client.from_("email_cache").update({
                "workflow_stage": WorkflowStage.ANALYZED.value,
                "ai_analysis": ai_analysis,
                "updated_at": datetime.now().isoformat()
            }).eq("id", email_id).execute()
            
            # Registrar ação
            self._log_workflow_action(
                email_id=email_id,
                user_id=user_id,
                action="analyze",
                from_stage=WorkflowStage.RECEIVED,
                to_stage=WorkflowStage.ANALYZED,
                notes="Análise automática por IA concluída"
            )
            
            logger.info(f"📊 Email {email_id} analisado com sucesso")
            
            return {
                "success": True,
                "analysis": ai_analysis,
                "message": "Email analisado com sucesso"
            }
            
        except Exception as e:
            logger.error(f"Erro ao analisar email: {e}")
            return {"success": False, "error": str(e)}
    
    def approve_content(self, email_id: str, user_id: str, user_feedback: Dict) -> Dict:
        """Aprovar conteúdo gerado (segunda etapa)"""
        try:
            email = self.get_email_details(email_id)
            if not email:
                return {"success": False, "error": "Email não encontrado"}
            
            if email["workflow_stage"] != WorkflowStage.ANALYZED.value:
                return {"success": False, "error": "Email precisa estar analisado"}
            
            # Atualizar com feedback do usuário
            update_result = self.db.client.from_("email_cache").update({
                "workflow_stage": WorkflowStage.APPROVED_CONTENT.value,
                "user_feedback": user_feedback,
                "assigned_to": user_id,
                "updated_at": datetime.now().isoformat()
            }).eq("id", email_id).execute()
            
            # Registrar ação
            self._log_workflow_action(
                email_id=email_id,
                user_id=user_id,
                action="approve_content",
                from_stage=WorkflowStage.ANALYZED,
                to_stage=WorkflowStage.APPROVED_CONTENT,
                notes=user_feedback.get("notes", "Conteúdo aprovado pelo usuário"),
                metadata=user_feedback
            )
            
            logger.info(f"✅ Conteúdo do email {email_id} aprovado")
            
            return {"success": True, "message": "Conteúdo aprovado com sucesso"}
            
        except Exception as e:
            logger.error(f"Erro ao aprovar conteúdo: {e}")
            return {"success": False, "error": str(e)}
    
    def prepare_for_publish(self, email_id: str, user_id: str, publish_data: Dict) -> Dict:
        """Preparar para publicação (terceira etapa)"""
        try:
            email = self.get_email_details(email_id)
            if not email:
                return {"success": False, "error": "Email não encontrado"}
            
            if email["workflow_stage"] != WorkflowStage.APPROVED_CONTENT.value:
                return {"success": False, "error": "Conteúdo precisa estar aprovado"}
            
            # Preparar dados para publicação
            ai_analysis = email.get("ai_analysis", {})
            generated_content = ai_analysis.get("generated_content", {})
            
            # Merge com dados do usuário
            final_content = {
                **generated_content,
                **publish_data,
                "prepared_by": user_id,
                "prepared_at": datetime.now().isoformat()
            }
            
            # Atualizar email
            updated_analysis = {**ai_analysis, "final_content": final_content}
            
            update_result = self.db.client.from_("email_cache").update({
                "workflow_stage": WorkflowStage.READY_PUBLISH.value,
                "ai_analysis": updated_analysis,
                "updated_at": datetime.now().isoformat()
            }).eq("id", email_id).execute()
            
            # Registrar ação
            self._log_workflow_action(
                email_id=email_id,
                user_id=user_id,
                action="prepare_publish",
                from_stage=WorkflowStage.APPROVED_CONTENT,
                to_stage=WorkflowStage.READY_PUBLISH,
                notes="Preparado para publicação",
                metadata=publish_data
            )
            
            logger.info(f"📝 Email {email_id} preparado para publicação")
            
            return {"success": True, "message": "Email preparado para publicação"}
            
        except Exception as e:
            logger.error(f"Erro ao preparar para publicação: {e}")
            return {"success": False, "error": str(e)}
    
    def publish_to_wordpress(self, email_id: str, user_id: str) -> Dict:
        """Publicar no WordPress (etapa final)"""
        try:
            email = self.get_email_details(email_id)
            if not email:
                return {"success": False, "error": "Email não encontrado"}
            
            if email["workflow_stage"] != WorkflowStage.READY_PUBLISH.value:
                return {"success": False, "error": "Email não está pronto para publicação"}
            
            # Obter conteúdo final
            ai_analysis = email.get("ai_analysis", {})
            final_content = ai_analysis.get("final_content", {})
            
            if not final_content:
                return {"success": False, "error": "Conteúdo final não encontrado"}
            
            # Publicar no WordPress
            publish_result = wp_publisher.publish_post(
                title=final_content.get("titulo", email["subject"]),
                content=final_content.get("conteudo", ""),
                category=final_content.get("categoria"),
                tags=final_content.get("tags", []),
                meta_description=final_content.get("meta_descricao")
            )
            
            if publish_result.get("success"):
                # Atualizar email como publicado
                update_result = self.db.client.from_("email_cache").update({
                    "workflow_stage": WorkflowStage.PUBLISHED.value,
                    "wordpress_post_id": publish_result.get("post_id"),
                    "updated_at": datetime.now().isoformat()
                }).eq("id", email_id).execute()
                
                # Registrar ação
                self._log_workflow_action(
                    email_id=email_id,
                    user_id=user_id,
                    action="publish",
                    from_stage=WorkflowStage.READY_PUBLISH,
                    to_stage=WorkflowStage.PUBLISHED,
                    notes=f"Publicado no WordPress - Post ID: {publish_result.get('post_id')}",
                    metadata={"wordpress_post_id": publish_result.get("post_id")}
                )
                
                logger.info(f"🚀 Email {email_id} publicado no WordPress")
                
                return {
                    "success": True,
                    "message": "Email publicado com sucesso",
                    "wordpress_post_id": publish_result.get("post_id")
                }
            else:
                return {"success": False, "error": publish_result.get("error", "Erro na publicação")}
                
        except Exception as e:
            logger.error(f"Erro ao publicar no WordPress: {e}")
            return {"success": False, "error": str(e)}
    
    # ==========================================
    # AÇÕES DE CONTROLE
    # ==========================================
    
    def reject_email(self, email_id: str, user_id: str, reason: str) -> Dict:
        """Rejeitar email em qualquer etapa"""
        try:
            email = self.get_email_details(email_id)
            if not email:
                return {"success": False, "error": "Email não encontrado"}
            
            current_stage = email["workflow_stage"]
            
            # Atualizar como rejeitado
            update_result = self.db.client.from_("email_cache").update({
                "workflow_stage": WorkflowStage.REJECTED.value,
                "user_feedback": {"rejection_reason": reason, "rejected_at": datetime.now().isoformat()},
                "updated_at": datetime.now().isoformat()
            }).eq("id", email_id).execute()
            
            # Registrar ação
            self._log_workflow_action(
                email_id=email_id,
                user_id=user_id,
                action="reject",
                from_stage=WorkflowStage(current_stage),
                to_stage=WorkflowStage.REJECTED,
                notes=f"Rejeitado: {reason}"
            )
            
            logger.info(f"❌ Email {email_id} rejeitado")
            
            return {"success": True, "message": "Email rejeitado"}
            
        except Exception as e:
            logger.error(f"Erro ao rejeitar email: {e}")
            return {"success": False, "error": str(e)}
    
    def archive_email(self, email_id: str, user_id: str) -> Dict:
        """Arquivar email (para emails antigos)"""
        try:
            update_result = self.db.client.from_("email_cache").update({
                "workflow_stage": WorkflowStage.ARCHIVED.value,
                "is_auto_process": False,
                "updated_at": datetime.now().isoformat()
            }).eq("id", email_id).execute()
            
            # Registrar ação
            self._log_workflow_action(
                email_id=email_id,
                user_id=user_id,
                action="archive",
                from_stage=None,
                to_stage=WorkflowStage.ARCHIVED,
                notes="Email arquivado"
            )
            
            return {"success": True, "message": "Email arquivado"}
            
        except Exception as e:
            logger.error(f"Erro ao arquivar email: {e}")
            return {"success": False, "error": str(e)}
    
    def update_priority(self, email_id: str, user_id: str, priority: int) -> Dict:
        """Atualizar prioridade do email"""
        try:
            if priority not in [1, 2, 3]:
                return {"success": False, "error": "Prioridade deve ser 1, 2 ou 3"}
            
            update_result = self.db.client.from_("email_cache").update({
                "priority": priority,
                "updated_at": datetime.now().isoformat()
            }).eq("id", email_id).execute()
            
            priority_names = {1: "Alta", 2: "Média", 3: "Baixa"}
            
            # Registrar ação
            self._log_workflow_action(
                email_id=email_id,
                user_id=user_id,
                action="update_priority",
                from_stage=None,
                to_stage=None,
                notes=f"Prioridade alterada para {priority_names[priority]}"
            )
            
            return {"success": True, "message": f"Prioridade alterada para {priority_names[priority]}"}
            
        except Exception as e:
            logger.error(f"Erro ao atualizar prioridade: {e}")
            return {"success": False, "error": str(e)}
    
    def assign_email(self, email_id: str, assigned_to: str, user_id: str) -> Dict:
        """Atribuir email a um usuário"""
        try:
            update_result = self.db.client.from_("email_cache").update({
                "assigned_to": assigned_to,
                "updated_at": datetime.now().isoformat()
            }).eq("id", email_id).execute()
            
            # Registrar ação
            self._log_workflow_action(
                email_id=email_id,
                user_id=user_id,
                action="assign",
                from_stage=None,
                to_stage=None,
                notes=f"Email atribuído ao usuário {assigned_to}"
            )
            
            return {"success": True, "message": "Email atribuído com sucesso"}
            
        except Exception as e:
            logger.error(f"Erro ao atribuir email: {e}")
            return {"success": False, "error": str(e)}
    
    # ==========================================
    # HELPERS
    # ==========================================
    
    def _log_workflow_action(self, email_id: str, user_id: str, action: str, 
                           from_stage: Optional[WorkflowStage], to_stage: Optional[WorkflowStage],
                           notes: Optional[str] = None, metadata: Optional[Dict] = None):
        """Registrar ação no histórico do workflow"""
        try:
            self.db.client.rpc('log_workflow_action', {
                'p_email_id': email_id,
                'p_user_id': user_id,
                'p_action': action,
                'p_from_stage': from_stage.value if from_stage else None,
                'p_to_stage': to_stage.value if to_stage else None,
                'p_notes': notes,
                'p_metadata': metadata
            }).execute()
        except Exception as e:
            logger.error(f"Erro ao registrar ação do workflow: {e}")

# Instância global
email_workflow = EmailWorkflowManager() 

File: frontend/js/email-workflow.js
/**
 * Email Workflow Module - Gerencia o fluxo de aprovação de emails
 * Integra com o backend para controlar as etapas do workflow
 */

class EmailWorkflowManager {
    constructor() {
        this.currentEmails = [];
        this.currentFilters = {};
        this.selectedEmail = null;
        
        console.log('📧 Email Workflow Manager inicializado');
    }
    
    // ==========================================
    // INITIALIZATION
    // ==========================================
    
    init() {
        this.setupEventListeners();
        this.loadWorkflowStats();
        this.loadEmailsList();
    }
    
    setupEventListeners() {
        // Filtros
        document.getElementById('emailStatusFilter')?.addEventListener('change', () => {
            this.applyFilters();
        });
        
        document.getElementById('emailPriorityFilter')?.addEventListener('change', () => {
            this.applyFilters();
        });
        
        document.getElementById('emailTypeFilter')?.addEventListener('change', () => {
            this.applyFilters();
        });
        
        // Botão de atualizar
        document.getElementById('refreshEmailsButton')?.addEventListener('click', () => {
            this.refreshData();
        });
    }
    
    // ==========================================
    // DATA LOADING
    // ==========================================
    
    async loadWorkflowStats() {
        try {
            const response = await authManager.apiCall('/workflow/dashboard');
            if (response.ok) {
                const data = await response.json();
                this.updateStatsUI(data.stats);
            }
        } catch (error) {
            console.error('Erro ao carregar estatísticas do workflow:', error);
        }
    }
    
    async loadEmailsList(filters = {}) {
        try {
            const params = new URLSearchParams({
                limit: '50',
                offset: '0',
                ...filters
            });
            
            const response = await authManager.apiCall(`/workflow/emails?${params}`);
            if (response.ok) {
                const data = await response.json();
                this.currentEmails = data.emails;
                this.renderEmailsTable(data.emails);
            }
        } catch (error) {
            console.error('Erro ao carregar lista de emails:', error);
            this.showError('Erro ao carregar emails');
        }
    }
    
    async loadEmailDetails(emailId) {
        try {
            const response = await authManager.apiCall(`/workflow/emails/${emailId}`);
            if (response.ok) {
                const data = await response.json();
                return data.email;
            }
        } catch (error) {
            console.error('Erro ao carregar detalhes do email:', error);
            return null;
        }
    }
    
    // ==========================================
    // UI UPDATES
    // ==========================================
    
    updateStatsUI(stats) {
        document.getElementById('pendingReview').textContent = stats.pending_review || 0;
        document.getElementById('readyToPublish').textContent = stats.ready_to_publish || 0;
        document.getElementById('publishedToday').textContent = stats.published_today || 0;
        document.getElementById('totalEmails').textContent = stats.total_emails || 0;
    }
    
    renderEmailsTable(emails) {
        const tableContainer = document.getElementById('emailsTable');
        if (!tableContainer) return;
        
        if (emails.length === 0) {
            tableContainer.innerHTML = `
                <div class="p-8 text-center text-gray-500">
                    <i data-lucide="inbox" class="h-12 w-12 mx-auto mb-4 text-gray-300"></i>
                    <p>Nenhum email encontrado</p>
                </div>
            `;
            return;
        }
        
        const table = `
            <table class="min-w-full divide-y divide-gray-200">
                <thead class="bg-gray-50">
                    <tr>
                        <th class="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider">
                            Email
                        </th>
                        <th class="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider">
                            Status
                        </th>
                        <th class="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider">
                            Prioridade
                        </th>
                        <th class="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider">
                            Recebido
                        </th>
                        <th class="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider">
                            Ações
                        </th>
                    </tr>
                </thead>
                <tbody class="bg-white divide-y divide-gray-200">
                    ${emails.map(email => this.renderEmailRow(email)).join('')}
                </tbody>
            </table>
        `;
        
        tableContainer.innerHTML = table;
        
        // Reprocessar ícones do Lucide
        if (typeof lucide !== 'undefined') {
            lucide.createIcons();
        }
        
        // Adicionar event listeners para ações
        this.setupEmailActions();
    }
    
    renderEmailRow(email) {
        const priorityClass = this.getPriorityClass(email.priority);
        const statusBadge = this.getStatusBadge(email.workflow_stage);
        const receivedDate = new Date(email.received_at).toLocaleDateString('pt-BR');
        
        return `
            <tr class="hover:bg-gray-50">
                <td class="px-6 py-4 whitespace-nowrap">
                    <div class="flex items-center">
                        <div class="flex-shrink-0 h-10 w-10">
                            <div class="h-10 w-10 rounded-full bg-gray-100 flex items-center justify-center">
                                <i data-lucide="mail" class="h-5 w-5 text-gray-600"></i>
                            </div>
                        </div>
                        <div class="ml-4">
                            <div class="text-sm font-medium text-gray-900 truncate max-w-xs">
                                ${email.subject}
                            </div>
                            <div class="text-sm text-gray-500">
                                De: ${email.sender}
                            </div>
                        </div>
                    </div>
                </td>
                <td class="px-6 py-4 whitespace-nowrap">
                    ${statusBadge}
                </td>
                <td class="px-6 py-4 whitespace-nowrap">
                    <span class="inline-flex items-center px-2.5 py-0.5 rounded-full text-xs font-medium ${priorityClass}">
                        ${this.getPriorityText(email.priority)}
                    </span>
                </td>
                <td class="px-6 py-4 whitespace-nowrap text-sm text-gray-500">
                    ${receivedDate}
                </td>
                <td class="px-6 py-4 whitespace-nowrap text-sm font-medium">
                    <div class="flex space-x-2">
                        <button 
                            onclick="emailWorkflow.viewEmail('${email.id}')" 
                            class="text-blue-600 hover:text-blue-900"
                            title="Ver detalhes"
                        >
                            <i data-lucide="eye" class="h-4 w-4"></i>
                        </button>
                        ${this.getEmailActionButtons(email)}
                    </div>
                </td>
            </tr>
        `;
    }
    
    getStatusBadge(stage) {
        const badges = {
            'received': '<span class="inline-flex items-center px-2.5 py-0.5 rounded-full text-xs font-medium bg-gray-100 text-gray-800">Recebido</span>',
            'analyzed': '<span class="inline-flex items-center px-2.5 py-0.5 rounded-full text-xs font-medium bg-yellow-100 text-yellow-800">Analisado</span>',
            'approved_content': '<span class="inline-flex items-center px-2.5 py-0.5 rounded-full text-xs font-medium bg-blue-100 text-blue-800">Aprovado</span>',
            'ready_publish': '<span class="inline-flex items-center px-2.5 py-0.5 rounded-full text-xs font-medium bg-purple-100 text-purple-800">Pronto</span>',
            'published': '<span class="inline-flex items-center px-2.5 py-0.5 rounded-full text-xs font-medium bg-green-100 text-green-800">Publicado</span>',
            'rejected': '<span class="inline-flex items-center px-2.5 py-0.5 rounded-full text-xs font-medium bg-red-100 text-red-800">Rejeitado</span>',
            'archived': '<span class="inline-flex items-center px-2.5 py-0.5 rounded-full text-xs font-medium bg-gray-100 text-gray-600">Arquivado</span>'
        };
        
        return badges[stage] || badges['received'];
    }
    
    getPriorityClass(priority) {
        const classes = {
            1: 'bg-red-100 text-red-800',
            2: 'bg-yellow-100 text-yellow-800',
            3: 'bg-green-100 text-green-800'
        };
        return classes[priority] || classes[3];
    }
    
    getPriorityText(priority) {
        const texts = {
            1: 'Alta',
            2: 'Média',
            3: 'Baixa'
        };
        return texts[priority] || 'Baixa';
    }
    
    getEmailActionButtons(email) {
        const stage = email.workflow_stage;
        let buttons = '';
        
        switch (stage) {
            case 'received':
                buttons = `
                    <button 
                        onclick="emailWorkflow.analyzeEmail('${email.id}')" 
                        class="text-green-600 hover:text-green-900"
                        title="Analisar com IA"
                    >
                        <i data-lucide="brain" class="h-4 w-4"></i>
                    </button>
                `;
                break;
                
            case 'analyzed':
                buttons = `
                    <button 
                        onclick="emailWorkflow.approveContent('${email.id}')" 
                        class="text-green-600 hover:text-green-900"
                        title="Aprovar conteúdo"
                    >
                        <i data-lucide="check" class="h-4 w-4"></i>
                    </button>
                `;
                break;
                
            case 'approved_content':
                buttons = `
                    <button 
                        onclick="emailWorkflow.preparePublish('${email.id}')" 
                        class="text-blue-600 hover:text-blue-900"
                        title="Preparar publicação"
                    >
                        <i data-lucide="edit" class="h-4 w-4"></i>
                    </button>
                `;
                break;
                
            case 'ready_publish':
                buttons = `
                    <button 
                        onclick="emailWorkflow.publishEmail('${email.id}')" 
                        class="text-purple-600 hover:text-purple-900"
                        title="Publicar no WordPress"
                    >
                        <i data-lucide="send" class="h-4 w-4"></i>
                    </button>
                `;
                break;
        }
        
        // Botão de rejeitar (disponível em qualquer etapa exceto publicado)
        if (!['published', 'rejected', 'archived'].includes(stage)) {
            buttons += `
                <button 
                    onclick="emailWorkflow.rejectEmail('${email.id}')" 
                    class="text-red-600 hover:text-red-900"
                    title="Rejeitar email"
                >
                    <i data-lucide="x" class="h-4 w-4"></i>
                </button>
            `;
        }
        
        return buttons;
    }
    
    setupEmailActions() {
        // Event listeners já são configurados via onclick nos botões
        // Mas podemos adicionar outros eventos aqui se necessário
    }
    
    // ==========================================
    // WORKFLOW ACTIONS
    // ==========================================
    
    async analyzeEmail(emailId) {
        try {
            this.showLoading(true);
            
            const response = await authManager.apiCall(`/workflow/emails/${emailId}/analyze`, {
                method: 'POST'
            });
            
            if (response.ok) {
                const result = await response.json();
                this.showSuccess('Email analisado com sucesso!');
                this.refreshData();
            } else {
                const error = await response.json();
                this.showError(error.detail || 'Erro ao analisar email');
            }
        } catch (error) {
            console.error('Erro ao analisar email:', error);
            this.showError('Erro ao analisar email');
        } finally {
            this.showLoading(false);
        }
    }
    
    async approveContent(emailId) {
        // Aqui poderia abrir um modal para feedback detalhado
        // Por simplicidade, vamos aprovar diretamente
        try {
            this.showLoading(true);
            
            const response = await authManager.apiCall(`/workflow/emails/${emailId}/approve`, {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json'
                },
                body: JSON.stringify({
                    rating: 5,
                    notes: 'Conteúdo aprovado via interface'
                })
            });
            
            if (response.ok) {
                this.showSuccess('Conteúdo aprovado com sucesso!');
                this.refreshData();
            } else {
                const error = await response.json();
                this.showError(error.detail || 'Erro ao aprovar conteúdo');
            }
        } catch (error) {
            console.error('Erro ao aprovar conteúdo:', error);
            this.showError('Erro ao aprovar conteúdo');
        } finally {
            this.showLoading(false);
        }
    }
    
    async preparePublish(emailId) {
        // Aqui deveria abrir um modal para editar o conteúdo final
        // Por simplicidade, vamos usar dados básicos
        try {
            const email = await this.loadEmailDetails(emailId);
            if (!email) return;
            
            const aiAnalysis = email.ai_analysis || {};
            const generatedContent = aiAnalysis.generated_content || {};
            
            this.showLoading(true);
            
            const response = await authManager.apiCall(`/workflow/emails/${emailId}/prepare`, {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json'
                },
                body: JSON.stringify({
                    titulo: generatedContent.titulo || email.subject,
                    conteudo: generatedContent.conteudo || 'Conteúdo a ser definido',
                    categoria: generatedContent.categoria || 'geral',
                    tags: generatedContent.tags || [],
                    meta_descricao: generatedContent.meta_descricao || ''
                })
            });
            
            if (response.ok) {
                this.showSuccess('Email preparado para publicação!');
                this.refreshData();
            } else {
                const error = await response.json();
                this.showError(error.detail || 'Erro ao preparar publicação');
            }
        } catch (error) {
            console.error('Erro ao preparar publicação:', error);
            this.showError('Erro ao preparar publicação');
        } finally {
            this.showLoading(false);
        }
    }
    
    async publishEmail(emailId) {
        if (!confirm('Tem certeza que deseja publicar este email no WordPress?')) {
            return;
        }
        
        try {
            this.showLoading(true);
            
            const response = await authManager.apiCall(`/workflow/emails/${emailId}/publish`, {
                method: 'POST'
            });
            
            if (response.ok) {
                const result = await response.json();
                this.showSuccess(`Email publicado com sucesso! Post ID: ${result.wordpress_post_id}`);
                this.refreshData();
            } else {
                const error = await response.json();
                this.showError(error.detail || 'Erro ao publicar email');
            }
        } catch (error) {
            console.error('Erro ao publicar email:', error);
            this.showError('Erro ao publicar email');
        } finally {
            this.showLoading(false);
        }
    }
    
    async rejectEmail(emailId) {
        const reason = prompt('Motivo da rejeição:');
        if (!reason) return;
        
        try {
            this.showLoading(true);
            
            const response = await authManager.apiCall(`/workflow/emails/${emailId}/reject`, {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json'
                },
                body: JSON.stringify({
                    reason: reason
                })
            });
            
            if (response.ok) {
                this.showSuccess('Email rejeitado');
                this.refreshData();
            } else {
                const error = await response.json();
                this.showError(error.detail || 'Erro ao rejeitar email');
            }
        } catch (error) {
            console.error('Erro ao rejeitar email:', error);
            this.showError('Erro ao rejeitar email');
        } finally {
            this.showLoading(false);
        }
    }
    
    async viewEmail(emailId) {
        try {
            const email = await this.loadEmailDetails(emailId);
            if (email) {
                this.showEmailDetails(email);
            }
        } catch (error) {
            console.error('Erro ao visualizar email:', error);
            this.showError('Erro ao carregar detalhes do email');
        }
    }
    
    // ==========================================
    // FILTERS & REFRESH
    // ==========================================
    
    applyFilters() {
        const statusFilter = document.getElementById('emailStatusFilter').value;
        const priorityFilter = document.getElementById('emailPriorityFilter').value;
        const typeFilter = document.getElementById('emailTypeFilter').value;
        
        const filters = {};
        
        if (statusFilter) filters.stage = statusFilter;
        if (priorityFilter) filters.priority = priorityFilter;
        if (typeFilter) filters.is_auto_process = typeFilter;
        
        this.currentFilters = filters;
        this.loadEmailsList(filters);
    }
    
    async refreshData() {
        await Promise.all([
            this.loadWorkflowStats(),
            this.loadEmailsList(this.currentFilters)
        ]);
    }
    
    // ==========================================
    // MODALS & UI HELPERS
    // ==========================================
    
    showEmailDetails(email) {
        // Implementar modal de detalhes do email
        alert(`Email: ${email.subject}\nStatus: ${email.workflow_stage}\nDe: ${email.sender}`);
    }
    
    showSuccess(message) {
        // Implementar notificação de sucesso
        alert(`✅ ${message}`);
    }
    
    showError(message) {
        // Implementar notificação de erro
        alert(`❌ ${message}`);
    }
    
    showLoading(show = true) {
        // Implementar indicador de loading
        const button = document.getElementById('refreshEmailsButton');
        if (button) {
            button.disabled = show;
            if (show) {
                button.innerHTML = '<i data-lucide="loader-2" class="h-4 w-4 mr-1 inline animate-spin"></i> Carregando...';
            } else {
                button.innerHTML = '<i data-lucide="refresh-cw" class="h-4 w-4 mr-1 inline"></i> Atualizar';
            }
        }
    }
}

// Instância global
const emailWorkflow = new EmailWorkflowManager(); 